### PR TITLE
Install: Hide admin password #3299

### PR DESCRIPTION
**Pull request by carlangas159**
Install: Hide admin password #3299

Show / hide password  taken from https://github.com/xuyongsky123/inputShowPwd

waiting for an answer from the author

### CODE REVIEW

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 64
           

Complexity increasing per file
==============================
- main/install/js/inputShowPwd.js  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016&bid=18715425)

File: main/install/js/inputShowPwd.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	(function () {
+		//处理ie9不支持栅格键点击触发onpropertychange事件
+		if (navigator.appName == 'Microsoft Internet Explorer') {
+			if (navigator.appVersion.match(/9./i) == '9.') {
+				T.inputEle.keyup(function (event) {
+					var t = $(this);
+					var _btn = t.parent().children('.showEle');
+					
+					if (event.keyCode == 8) {
+						;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:33]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+inputShowPwd.prototype.init = function (ele) {
+	var T = this;
+	this.container = $('.' + ele);  //input包装容器
+	this.inputEle = this.container.children('input[type="password"]');   //input输入框
+	this.deleteEle = this.container.children('.showEle'); //显示按钮
+	(function () {
+		//处理ie9不支持栅格键点击触发onpropertychange事件
+		if (navigator.appName == 'Microsoft Internet Explorer') {
+			if (navigator.appVersion.match(/9./i) == '9.') {
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:27]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	//初始化样式及效果
+	this.init(ele);
+	return document.getElementsByClassName(ele);
+}
+
+inputShowPwd.prototype.init = function (ele) {
+	var T = this;
+	this.container = $('.' + ele);  //input包装容器
+	this.inputEle = this.container.children('input[type="password"]');   //input输入框
+	this.deleteEle = this.container.children('.showEle'); //显示按钮
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:23]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	/*line-height: 36px;*/ /*控制input框文字的高度可以同时结合height来重写input框文字的高度*/
+	padding-left: 10px;
+	padding-right: 36px;
+	padding-right: 36px !important;
+}
+
+.inputShowPwd > .showEle {
+	background: url(image/pwd.png) 50% 50% no-repeat;
+	background-size: 24px 24px;
+	cursor: pointer;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:45]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		}).on('input propertychange', function () {
+			var t = $(this);
+			var _btn = t.parent().children('.showEle');
+			;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
+		});
+		T.deleteEle.on('mousedown', function () {
+			var _parent = $(this).parent();
+			var _pwd = _parent.children('input[type="password"]');
+			_pwd.hide();
+			_parent.append($(T.createNewInput(_pwd.get(0), 'text')).show());
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:51]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*
+*
+* */
+function inputShowPwd(ele) {
+	this.createNewInput = function (oldObject, oType) {
+		var newObject = document.createElement('input');
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:6]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	display: none;
+	height: 24px;
+	margin-top: -12px;
+	position: absolute;
+	right: 15px;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:50]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js (hunk 2 above)
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:27]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+		T.inputEle.on('focus', function () {
+			var t = $(this);
+			var _btn = t.parent().children('.showEle');
+			;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
+		}).on('input propertychange', function () {
+			var t = $(this);
+			var _btn = t.parent().children('.showEle');
+			;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
+		});
+		T.deleteEle.on('mousedown', function () {
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:47]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+	padding-right: 36px;
+	padding-right: 36px !important;
+}
+
+.inputShowPwd > .showEle {
+	background: url(image/pwd.png) 50% 50% no-repeat;
+	background-size: 24px 24px;
+	cursor: pointer;
+	display: none;
+	height: 24px;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:47]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 10]
```diff
@@ -0,0 +1,58 @@
+* {
+	box-sizing: border-box;
+	font-size: 16px;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:3]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+	position: relative;
+}
+
+.inputShowPwd > input {
+	/*width: auto; 
+	height: auto;*/ /*默认input的宽高可以同时结合width\height来重写input框的宽度/高度*/
+	/*line-height: 36px;*/ /*控制input框文字的高度可以同时结合height来重写input框文字的高度*/
+	padding-left: 10px;
+	padding-right: 36px;
+	padding-right: 36px !important;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:39]: Codacy found an issue: [Unexpected duplicate "36px" (declaration-block-no-duplicate-properties)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
+/*
+*
+* */
+function inputShowPwd(ele) {
+	this.createNewInput = function (oldObject, oType) {
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:5]: Codacy found an issue: [Function has a complexity of 5. Maximum allowed is 4.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+	display: none;
+}
+
+input {
+	box-sizing: border-box;
+}
+
+input:active, input:focus {
+	-moz-outline: none;
+	outline: none;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:18]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+		});
+		T.deleteEle.on('mousedown', function () {
+			var _parent = $(this).parent();
+			var _pwd = _parent.children('input[type="password"]');
+			_pwd.hide();
+			_parent.append($(T.createNewInput(_pwd.get(0), 'text')).show());
+		}).on('mouseup mouseout', function () {
+			var _parent = $(this).parent();
+			_parent.find('input[type="text"]').remove();
+			_parent.children('input[type="password"]').show();
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:55]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+/*去掉移动端firefox 按钮获得焦点显示虚线框的样式 */
+/* main style */
+.inputShowPwd {
+	display: inline-block;
+	position: relative;
+}
+
+.inputShowPwd > input {
+	/*width: auto; 
+	height: auto;*/ /*默认input的宽高可以同时结合width\height来重写input框的宽度/高度*/
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:35]: Codacy found an issue: [Expected whitespace before "*/" (comment-whitespace-inside)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 16]
```diff
@@ -0,0 +1,58 @@
+* {
+	box-sizing: border-box;
+	font-size: 16px;
+	vertical-align: middle;
+}
+
+/* 清除现代IE浏览器的输入框和密码框的默认交互按钮 */
+::-ms-clear, ::-ms-reveal {
+	display: none;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:9]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+.inputShowPwd > .showEle {
+	background: url(image/pwd.png) 50% 50% no-repeat;
+	background-size: 24px 24px;
+	cursor: pointer;
+	display: none;
+	height: 24px;
+	margin-top: -12px;
+	position: absolute;
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:49]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+/* main style */
+.inputShowPwd {
+	display: inline-block;
+	position: relative;
+}
+
+.inputShowPwd > input {
+	/*width: auto; 
+	height: auto;*/ /*默认input的宽高可以同时结合width\height来重写input框的宽度/高度*/
+	/*line-height: 36px;*/ /*控制input框文字的高度可以同时结合height来重写input框文字的高度*/
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:36]: Codacy found an issue: [Expected empty line before comment (comment-empty-line-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+					var t = $(this);
+					var _btn = t.parent().children('.showEle');
+					
+					if (event.keyCode == 8) {
+						;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
+					}
+				});
+			}
+		}
+		T.inputEle.on('focus', function () {
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:38]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+			var t = $(this);
+			var _btn = t.parent().children('.showEle');
+			;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
+		}).on('input propertychange', function () {
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:42]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+	outline: none;
+}
+
+/* PC chrome文本输入框的边框焦点时候的颜色 */
+input[type="button"]::-moz-focus-inner, input[type="submit"]::-moz-focus-inner {
+	border: none;
+}
+
+/*去掉移动端firefox 按钮获得焦点显示虚线框的样式 */
+/* main style */
```
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:27]: Codacy found an issue: [Expected empty line before comment (comment-empty-line-before)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/css/inputShowPwd.css (hunk 11 above)
Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:39]: Codacy found an issue: [Use of !important](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

Comment (jmontoyaa) [inline main/install/css/inputShowPwd.css:39]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/install/js/inputShowPwd.js [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+			var t = $(this);
+			var _btn = t.parent().children('.showEle');
+			;(t.val() == '') ? _btn.fadeOut(100) : _btn.fadeIn(100);
+		});
+		T.deleteEle.on('mousedown', function () {
+			var _parent = $(this).parent();
+			var _pwd = _parent.children('input[type="password"]');
+			_pwd.hide();
+			_parent.append($(T.createNewInput(_pwd.get(0), 'text')).show());
+		}).on('mouseup mouseout', function () {
```
Comment (jmontoyaa) [inline main/install/js/inputShowPwd.js:52]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5720016)

File: main/course_info/delete_course.php [hunk 23]
```diff
@@ -38,7 +38,7 @@
 } else {
     $message = '<h3>'.get_lang('Course').' : '.$current_course_name.' ('.$current_course_code.') </h3>';
     $message .= '<p>'.get_lang('ByDel').'</p>';
-    $message .= '<p><a class="btn btn-primary" 
+    $message .= '<p><a class="btn btn-primary"
```
Comment (AngelFQC) [inline main/course_info/delete_course.php:41]: Undo changes in this file. File not related to issue.

File: main/install/index.php [hunk 24]
```diff
@@ -963,4 +965,9 @@ function send_contact_information() {
     </div>
   </div>
 </body>
+<script>
+    $(function(){
+        var _input = new inputShowPwd('inputShowPwd');
+    });
+</script>
```
Comment (AngelFQC) [inline main/install/index.php:972]: No es necesario agregar otra librería JS para hacer el cambio. Por ejemplo aquí:
https://github.com/chamilo/mobile/blob/master/www/src/view/login.js#L69-L72

**Comment by ywarnier** [comment]
OK, anular este cambio entonces y hacerlo por la forma sugerida por @AngelFQC en el código de la app móvil (seguir enlace).
En este caso no creo que muestre un icono de ojo ni nada, así que quizás haya algo más que hacer para que este evento suceda solo al dar clic al icono de ojo. Y quedará mejor por no incluir una librería externa (el ejemplo es puro jQuery).

**Comment by ywarnier** [comment]
Cierro el PR pero la tarea #3299 queda por hacer.

**Pull request closed without merging**